File: sdk/typescript/src/portfolio.ts
/**
 * TypeScript SDK Client for Enterprise AI Portfolio Platform (Phase 5.28).
 */

export class PortfolioClient {
  private client: any;

  constructor(client: any) {
    this.client = client;
  }

  async createStrategy(name: string, description: string, horizon: string = 'NEAR_TERM'): Promise<any> {
    return this.client.request('POST', '/v1/portfolio/strategies', { name, description, horizon });
  }

  async listStrategies(): Promise<any[]> {
    return this.client.request('GET', '/v1/portfolio/strategies');
  }

  async discoverOpportunity(title: string, description: string): Promise<any> {

    return this.client.request('POST', '/v1/portfolio/opportunities', { title, description });
  }

  async createInitiative(title: string, description: string): Promise<any> {
    return this.client.request('POST', '/v1/portfolio/initiatives', { title, description });
  }

  async createBusinessCase(initiativeId: string, problemStatement: string): Promise<any> {
    return this.client.request('POST', '/v1/portfolio/business-cases', { initiative_id: initiativeId, problem_statement: problemStatement });
  }

  async prioritize(): Promise<any> {
    return this.client.request('POST', '/v1/portfolio/prioritize');
  }

  async optimize(): Promise<any> {
    return this.client.request('POST', '/v1/portfolio/optimize');
  }

  async proposeInvestment(initiativeId: string, amountUsd: number, riskLevel: string = 'HIGH'): Promise<any> {
    return this.client.request('POST', '/v1/portfolio/investments', { initiative_id: initiativeId, amount_usd: amountUsd, risk_level: riskLevel });
  }

  async allocateFunding(initiativeId: string, idempotencyKey: string, amountUsd: number): Promise<any> {
    return this.client.request('POST', '/v1/portfolio/funding/allocate', { initiative_id: initiativeId, idempotency_key: idempotencyKey, amount_usd: amountUsd });
  }

  async getAnalytics(): Promise<any> {
    return this.client.request('GET', '/v1/portfolio/analytics');
  }
}

File: sdk/go/portfolio.go
package llmengine

import (
	"context"
)

type PortfolioClient struct {
	client *Client
}

func NewPortfolioClient(client *Client) *PortfolioClient {
	return &PortfolioClient{client: client}
}

func (c *PortfolioClient) CreateStrategy(ctx context.Context, name, description, horizon string) (map[string]interface{}, error) {
	payload := map[string]interface{}{
		"name":        name,
		"description": description,
		"horizon":     horizon,
	}
	var res map[string]interface{}
	err := c.client.Post(ctx, "/v1/portfolio/strategies", payload, &res)
	return res, err
}

func (c *PortfolioClient) ListStrategies(ctx context.Context) ([]map[string]interface{}, error) {
	var res []map[string]interface{}
	err := c.client.Get(ctx, "/v1/portfolio/strategies", &res)
	return res, err
}

func (c *PortfolioClient) DiscoverOpportunity(ctx context.Context, title, description string) (map[string]interface{}, error) {
	payload := map[string]interface{}{
		"title":       title,
		"description": description,
	}
	var res map[string]interface{}
	err := c.client.Post(ctx, "/v1/portfolio/opportunities", payload, &res)
	return res, err
}

func (c *PortfolioClient) CreateInitiative(ctx context.Context, title, description string) (map[string]interface{}, error) {
	payload := map[string]interface{}{
		"title":       title,
		"description": description,
	}
	var res map[string]interface{}
	err := c.client.Post(ctx, "/v1/portfolio/initiatives", payload, &res)
	return res, err
}

func (c *PortfolioClient) Prioritize(ctx context.Context) (map[string]interface{}, error) {
	var res map[string]interface{}
	err := c.client.Post(ctx, "/v1/portfolio/prioritize", nil, &res)
	return res, err
}

func (c *PortfolioClient) Optimize(ctx context.Context) (map[string]interface{}, error) {
	var res map[string]interface{}
	err := c.client.Post(ctx, "/v1/portfolio/optimize", nil, &res)
	return res, err
}

func (c *PortfolioClient) GetAnalytics(ctx context.Context) (map[string]interface{}, error) {
	var res map[string]interface{}
	err := c.client.Get(ctx, "/v1/portfolio/analytics", &res)
	return res, err
}
